### PR TITLE
Allow list entry to be submitted on enter key press - React Application

### PR DESCRIPTION
**Pull request by streetcoder123**
Task: https://microsoftgarage.visualstudio.com/Intern%20GitHub/_workitems/edit/26876
Github Issue: #570 

**Changes**
- React List Template should be able to submit a list item both by pressing the submit button and on enter key press.
- This is achieved using the onSubmit command in the form tag rather than using it in the submit button.

### CODE REVIEW

**Comment by SahilTara** [comment]
Hey,
Just wondering if you could also tackle angular's too in a seperate PR (or this one since related problem with same solution) since I suspect it will be a problem. https://github.com/microsoft/WebTemplateStudio/blob/dev/templates/Web/Pages/Angular.List/src/app/components/AngularList/ListForm/list-form.component.html

Syntax slightly different but shouldn't be too bad.

Edit: I would look into ngSubmit.

File: src/client/src/translations/whitelist_en.json [hunk 1]
```diff
@@ -68,7 +68,6 @@
   "cosmosResourceModule.saveChanges",
   "cosmosResourceModule.createCosmosRes",
   "cosmosResourceModule.internalName",
-  "footer.license",
```
Comment (jsondoo) [inline src/client/src/translations/whitelist_en.json:71]: why was this line removed?

Comment (SahilTara) [reply src/client/src/translations/whitelist_en.json:71]: I'm guessing this was due to running build since footer.license isnt used anywhere else but I may be wrong. This shouldn't be removed technically, but it reveals an underlying issue https://github.com/microsoft/WebTemplateStudio/blob/c3eeff139a294756156842a7cf394d254d500f34/src/client/src/containers/Footer/index.tsx
Looking at this file it seems that footer.license isnt used anywhere, it should be doing similar to what the aria label is doing with intl.

**Comment by streetcoder123** [comment]
> Hey,
> Just wondering if you could also tackle angular's too in a seperate PR (or this one since related problem with same solution) since I suspect it will be a problem. https://github.com/microsoft/WebTemplateStudio/blob/dev/templates/Web/Pages/Angular.List/src/app/components/AngularList/ListForm/list-form.component.html
> Syntax slightly different but shouldn't be too bad.
> Edit: I would look into ngSubmit.

Yes, I am creating a separate PR for angular. It should also be there soon.

**Comment by crutkas** [comment]
what is the actual issue on Github this relates to.

**Comment by jsondoo** [comment]
#570

File: src/client/src/translations/whitelist_en.json (hunk 1 above)
Comment (crutkas) [reply src/client/src/translations/whitelist_en.json:71]: @njindal239 why was the line removed

**Comment by crutkas** [comment]
why did the pipeline fail?  @sibille the logs are very spare here

**Comment by crutkas** [comment]
@njindal239  also looks like there is a branch conflict now

File: src/client/src/translations/whitelist_en.json (hunk 1 above)
Comment (streetcoder123) [reply src/client/src/translations/whitelist_en.json:71]: This change comes in when I run ./build on the project. I am not sure whether it needs to be there or not but since it came out of building the project, I committed it with my changes.

**Comment by sibille** [comment]
@crutkas, seems the build failed during initialization, went fine when I hit "Rebuild".

**Review by SahilTara — DISMISSED** [review]
Everything seems to work as intended for the list. Code looks sound. The line that was removed due to ./build needs an issue against it since we are not using intl properly for that one parameter. #614

**Review by piotrmark — APPROVED** [review]

File: templates/Web/Pages/React.List/src/components/ReactList/ListForm.jsx [hunk 2]
```diff
@@ -12,9 +12,9 @@ export default function ListForm(props) {
         placeholder="Add text here..."
         aria-label="Add text here..."
       />
-      <button className="btn btn-primary ml-2" onClick={onAddListItem}>
+      <button className="btn btn-primary ml-2">
```
Comment (piotrmark) [inline templates/Web/Pages/React.List/src/components/ReactList/ListForm.jsx]: I think it's better to explicitly define type="submit" on the button. It's default for most of the browsers, but per W3Schools: "Always specify the type attribute for the <button> element. Different browsers may use different default types for the <button> element."
https://www.w3schools.com/tags/att_button_type.asp

**Review by piotrmark — CHANGES_REQUESTED** [review]

File: templates/Web/Pages/React.List/src/components/ReactList/ListForm.jsx (hunk 2 above)
Comment (streetcoder123) [reply templates/Web/Pages/React.List/src/components/ReactList/ListForm.jsx]: That's a good catch. I will update that. Thanks!

**Review by jsondoo — APPROVED** [review]

**Review by piotrmark — APPROVED** [review]

**Comment by Tanya0609** [comment]
I did review. It works so approve this change

**Review by Tanya0609 — APPROVED** [review]

File: templates/Web/Pages/React.List/src/components/ReactList/ListForm.jsx (hunk 2 above)
Comment (SahilTara) [reply templates/Web/Pages/React.List/src/components/ReactList/ListForm.jsx]: Really good catch!
